pkgs/by-name/ig/igir/package.nix

```
@@ -2,46 +2,87 @@
   lib,
   buildNpmPackage,
   fetchFromGitHub,
-
-  # for patching bundled 7z binary from the 7zip-bin node module
-  # at lib/node_modules/igir/node_modules/7zip-bin/linux/x64/7za
-  autoPatchelfHook,
   stdenv,
+  autoPatchelfHook,
+  SDL2,
+  libuv,
+  lz4,
+  #tests
+  runCommand,
+  igir,
+  nodejs,
 }:
-
 buildNpmPackage rec {
   pname = "igir";
-  version = "2.11.0";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "emmercm";
     repo = "igir";
     rev = "v${version}";
-    hash = "sha256-NG0ZP8LOm7fZVecErTuLOfbp1yvXwHnwPkWTBzUJXWE=";
+    hash = "sha256-QPUnvirufY3UiSFVxX3xCmpAlzPKg5JxwAnVefdepqU=";
   };
 
-  npmDepsHash = "sha256-ADIEzr6PkGaJz27GKSVyTsrbz5zbud7BUb+OXPtP1Vo=";
+  npmDepsHash = "sha256-miXxMBtIqgxS4Ulhkq79VDCkBjDjoB8juPoDWE1UQUM=";
+
+  # This patch is only needed for 3.0.1 and needs to be removed afterwards
+  # along with the patch file itself
+  patches = [ ./0001-update-dependencies.patch ];
 
   # I have no clue why I have to do this
   postPatch = ''
     patchShebangs scripts/update-readme-help.sh
   '';
 
-  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [
+    (lib.getLib stdenv.cc.cc)
+    SDL2
+    libuv
+    lz4
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [ autoPatchelfHook ];
+
+  postFixup = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    # Fix chdman
+    find $out/lib/node_modules/igir/node_modules/@emmercm/ -name chdman -executable -type f \
+      -exec install_name_tool -change /opt/homebrew/opt/sdl2/lib/libSDL2-2.0.0.dylib ${SDL2}/lib/libSDL2-2.0.0.dylib {} \;
+
+    # Fix maxcso
+    find $out/lib/node_modules/igir/node_modules/@emmercm/ -name maxcso -executable -type f \
+      -exec install_name_tool -change /opt/homebrew/opt/libuv/lib/libuv.1.dylib ${libuv}/lib/libuv.1.dylib {} \;
+    find $out/lib/node_modules/igir/node_modules/@emmercm/ -name maxcso -executable -type f \
+      -exec install_name_tool -change /opt/homebrew/opt/lz4/lib/liblz4.1.dylib ${lz4.lib}/lib/liblz4.1.dylib {} \;
+  '';
```

Can we use the [fixDarwinDylibNames](https://github.com/NixOS/nixpkgs/blob/c79c52d6ddeccbdfa7366d7cbd732d59ee33c36f/pkgs/build-support/setup-hooks/fix-darwin-dylib-names.sh#L9-L11) hook for this?

```suggestion
  nativeBuildInputs =
    lib.optionals stdenv.hostPlatform.isLinux [ autoPatchelfHook ]
    ++ lib.optionals stdenv.hostPlatform.isDarwin [ fixDarwinDylibNames ];
```

---

Good question. Let me check.

---

Tried it, does not work. The comment on top of [fixDarwinDylibNames](https://github.com/NixOS/nixpkgs/blob/c79c52d6ddeccbdfa7366d7cbd732d59ee33c36f/pkgs/build-support/setup-hooks/fix-darwin-dylib-names.sh#L9-L11) also says:

> This setup hook fixes dylibs by setting their install names to their absolute path

However, we need to replace the absolute path inside the binaries compiled for homebrew to reference the corresponding libraries in nix.

---

Ah, it was worth a try. There might be a better way to do this, but this is probably fine for now.